src/loader.ts

```diff
@@ -316,5 +343,9 @@ async function resolveConfig(
     res.config = defu(envConfig, res.config);
   }
 
+  // Meta
+  res.meta = defu(res.sourceOptions.meta, res.config.$meta);
```

Comment:
I don't see any reason why we would overwrite the `meta` of the config.

```suggestion
  res.meta = res.config.$meta || {};
```

Reply:
Multi-app overrides for alias at least. As explained in PR notes, we don't necessarily need to advice using this feature but it is an advanced escape hatch.